internal/chain/synthetic_create_chain.go

```diff
@@ -91,10 +91,12 @@ func (SyntheticCreateChain) Validate(st *StateManager, tx *transactions.GenTrans
 			}
 
 			// Update the ADI's directory index
-			err = st.AddDirectoryEntry(u)
-			if err != nil {
-				return fmt.Errorf("failed to add ADI directory entry: %v", err)
-			}
+			if record.Header().SigSpecId != (types.Bytes32{}) {
```

Comment:
This would result in the key book never getting added to the directory index, because the key book's SigSpecId is always zero. If you check `n.GetDirectory(adiUrl)` in `TestCreateADI`, with this change the directory is missing `acc://RoadRunner/foo-book`.